### PR TITLE
SNOW-983098: redo arrow update 15.0.0

### PR DESCRIPTION
**Pull request by sfc-gh-ext-simba-hx**
teamwork issue 780
temporarily add build number in libsnowflakeclient version to avoid replacing artifacts on s3 built with gcc5. Will remove when ODBC also move to gcc8.

### CODE REVIEW

**Comment by sfc-gh-ext-simba-hx** [comment]
The SEGFAULT with Mac debug build is related to static regex defined in https://github.com/snowflakedb/libsnowflakeclient/blob/master/cpp/logger/SecretDetector.cpp when calling ~mem_block_cache at the ending of test application. Will get that fixed before asking for review.

**Comment by sfc-gh-ext-simba-hx** [comment]
Confirmed no performance issue with adding `-DBOOST_REGEX_MAX_CACHE_BLOCKS=0`
regex is used by arrow also for logging in libsfc directly.
Tested with ODBC driver using testapp in teamwork issue 697 (fetch a large result using SQLGetData) with both loglevel=6 (1GB log generated) and loglevel=0, compared with 3.3.0, no performance issue found.
new build vs 3.3.0 in seconds
loglevel=6:   94.043 / 112.014
loglevel=0:   7.344 / 7.401

File: include/snowflake/version.h [hunk 1]
```diff
@@ -5,6 +5,6 @@
 #ifndef SNOWFLAKE_CLIENT_VERSION_H
 #define SNOWFLAKE_CLIENT_VERSION_H
 
-#define SF_API_VERSION "1.0.10"
+#define SF_API_VERSION "1.0.10.1"
```
Comment (sfc-gh-dprzybysz) [inline include/snowflake/version.h:8]: why do we bump the version here? cannot we use 1.0.10?

Comment (sfc-gh-ext-simba-hx) [reply include/snowflake/version.h]: changed back.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Pull request merged**